packages/nextjs/components/scaffold-eth/Contract/MethodSelector.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from "react";
+import { Abi } from "viem";
+import { ChevronDownIcon, ChevronRightIcon, XMarkIcon } from "@heroicons/react/24/outline";
+
+type AbiFunction = Extract<Abi[number], { type: "function" }>;
+
+type AbiFallback = Extract<Abi[number], { type: "fallback" }>;
+
+type SelectableAbiType = AbiFunction | AbiFallback;
+
+interface MethodSelectorProps {
+  methodsWithInputs: SelectableAbiType[];
+  abi: Abi;
+  onMethodSelect: (selectedMethods: string) => void;
+  removeMethod: (methodName: string) => void;
+}
+
+export const MethodSelector = ({ methodsWithInputs, abi, onMethodSelect, removeMethod }: MethodSelectorProps) => {
+  const [isReadCollapsed, setIsReadCollapsed] = useState(false);
+  const [isWriteCollapsed, setIsWriteCollapsed] = useState(false);
+
+  const readMethods = methodsWithInputs.filter(
+    (method): method is AbiFunction => method.type === "function" && method.stateMutability === "view",
+  );
```

Comment:
I think the right condition should be : 

```ts
method.stateMutability === "view" || method.stateMutability === "pure"
```

Since pure functions can also have inputs 

PS: We can remove `method.type === "function"` this check since we are already doing that check in parent component and passing it here